core/pkg/service/flag-evaluation/connect_service.go

```diff
@@ -129,16 +130,28 @@ func (s *ConnectService) setupServer(svcConf service.Configuration) (net.Listene
 
 	path, handler := schemaConnectV1.NewServiceHandler(fes, append(svcConf.Options, marshalOpts)...)
 	mux.Handle(path, handler)
+	mux.Handle(grpchealth.NewHandler(grpchealth.NewStaticChecker(
+		schemaConnectV1.ServiceName,
+	)))
+	mux.Handle("/healthz", http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		w.WriteHeader(http.StatusOK)
+	}))
+	mux.Handle("/readyz", http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		if s.readinessEnabled && svcConf.ReadinessProbe() {
+			w.WriteHeader(http.StatusOK)
+		} else {
+			w.WriteHeader(http.StatusPreconditionFailed)
+		}
+	}))
 
 	s.serverMtx.Lock()
 	s.server = &http.Server{
 		ReadHeaderTimeout: time.Second,
-		Handler:           handler,
+		Handler:           mux,
```

Comment:
The `mux` created above was actually unused - instead we just used the one handler originally.

Instead of adding some conditions based on paths our other ready handler, I've used the `mux` which simplifies things a bit (this is what mux is for).